### ISSUE TITLE
Rule Change: Update `no-obj-calls` to report `Temporal`

### ISSUE THREAD

**Issue by Pixel998** [eslint/eslint#20673]
### What rule do you want to change?

no-obj-calls

### What change do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
Temporal();

new Temporal();
```

### What does the rule currently do for this code?

It does not report any errors.

### What will the rule do after it's changed?

It will report `Temporal` as a non-callable global namespace object, similar to `Math`, `JSON`, `Reflect`, `Atomics`, and `Intl`.

### Participation

- [x] I am willing to submit a pull request to implement this change.

### AI acknowledgment

- [x] I did not use AI to generate this issue report.
- [ ] (If the above is not checked) I have reviewed the AI-generated content before submitting.

### Additional comments

[Temporal](https://github.com/tc39/proposal-temporal) has reached TC39 stage 4, so we should support it in core rules.

**Comment by mdjermanovic** [comment]
Accepted, PR is welcome.
